lc-0226-invert-binary-tree/src/test/java/solution/CustomCsvArgTest.java

```diff
@@ -0,0 +1,182 @@
+package solution;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+
+public class CustomCsvArgTest {
+
+    static class MyTreeAsList {
+        public ArrayList<Integer> values;
+
+        // Generate List from delimeted CSV input string:
+        MyTreeAsList(String s) {
+            // Convert to String[] array:
+            s = s.replaceAll("\\[", "");
+            s = s.replaceAll("\\]", "");
+            String[] tokens = s.split(",");
+
+            // Generate ArrayList<>():
+            values = new ArrayList<>();
+            for (String token : tokens) {
+                // Account for 'null' and "" strings:
+                if (!token.equals("null") && !token.equals("")) {
+                    values.add(Integer.parseInt(token));
+                } else {
+                    values.add(null);
+                }
+            }
+        }
+
+        // Generate List from TreeNode head:
+        MyTreeAsList(TreeNode head) {
+
+            values = new ArrayList<>();
+
+            if (head == null) {
+                values.add(null); // Add a 'null' instead of having empty List to stay consistent
+            } else {
+                Queue<TreeNode> q = new LinkedList<>();
+                q.offer(head);
+
+                // Build arraylist:
+                while (!q.isEmpty()) {
+                    TreeNode curr = q.poll();
+                    if (curr != null) {
+                        values.add(curr.val);
+                        q.offer(curr.left);
+                        q.offer(curr.right);
+                    } else {
+                        values.add(null);
+                    }
+                }
+
+                // Trim tail nulls:
+                while (values.get(values.size() - 1) == null) {
+                    values.remove(values.size() - 1);
+                }
+            }
+        }
+    }
+
+    // Generate TreeNode tree and store it's 'head':
+    static class MyTree {
+        private TreeNode head;
+
+        MyTree(String s) {
+
+            // Use MyIntArrList to first build the List<Integer>
+            List<Integer> values = new MyTreeAsList(s).values;
```

Comment:
This:
```
= new MyTreeAsList(s).values;
```
can almost always can be replaced with:
```
= TestUtils.toList(s);
```

In other words, if you only constuct a class (e.g. `MyTreeAsList`) to get its guts and throw it away, you don't need that class. A function will be more appropriate.